hs_restclient/endpoints/resources.py

```diff
@@ -0,0 +1,237 @@
+            url_base=self.hs.url_base,
+            pid=self.pid)
+        r = self.hs._request('POST', url, None, payload)
+        return r
+
+
+class ResourceEndpoint(BaseEndpoint):
+    def __init__(self, hs, pid):
+        super(ResourceEndpoint, self).__init__(hs)
+        self.pid = pid
+        self.scimeta = ScimetaSubEndpoint(hs, pid)
+        self.functions = FunctionsSubEndpoint(hs, pid)
+
+    def copy(self):
+        """
+        Creates a copy of a resource
+
+        :return: string resource id
+        """
+        url = "{url_base}/resource/{pid}/copy/".format(url_base=self.hs.url_base,
+                                                       pid=self.pid)
+        r = self.hs._request('POST', url)
+        return r
+
+    def flag(self, payload):
+        """
+        Sets a single flag on a resource
+
+        :param payload:
+            t: can be one of make_public, make_private, make_shareable,
+            make_not_shareable, make_discoverable, make_not_discoverable
+        :return:
+            empty but with 202 status_code
+        """
+        url = "{url_base}/resource/{pid}/flag/".format(url_base=self.hs.url_base,
+                                                       pid=self.pid)
+
+        r = self.hs._request('POST', url, None, payload)
+        return r
+
+    def files(self, payload):
+        """
+        Uploads a file to a hydroshare resource
+
+        :param payload:
+            file: File object to upload to server
+            folder: folder path to upload the file to
+        :return: json object
+            resource_id: string resource id,
+            file_name: string name of file
+        """
+        url = "{url_base}/resource/{pid}/files/".format(url_base=self.hs.url_base,
+                                                       pid=self.pid)
+
+        params = {}
+        file_to_upload = self.hs._prepareFileForUpload(params,
+                                                       payload['file'],
+                                                       os.path.basename(payload['file']))
+
+        encoder = MultipartEncoder(params)
+        monitor = MultipartEncoderMonitor(encoder, default_progress_callback)
+
+        payload.pop('file')
+        r = self.hs._request('POST', url, None, data=monitor, headers={'Content-Type': monitor.content_type})
+        return r.text
+
+    def version(self):
+        """
+        Creates a new version of a resource
+
+        :return: resource id (string)
+        """
+        url = "{url_base}/resource/{pid}/version/".format(url_base=self.hs.url_base,
+                                                          pid=self.pid)
+        r = self.hs._request('POST', url)
+        return r
+
+
+class ResourceList(BaseEndpoint):
+    def __init__(self, hs, **kwargs):
+        super(ResourceList, self).__init__(hs)
+
+        """
+        Query the GET /hsapi/resource/ REST end point of the HydroShare server.
+

```

Comment:
Should we be including other parameters as per the validator here (https://github.com/hydroshare/hydroshare/blob/develop/hs_core/views/serializers.py#L69)?